application/src/main/java/org/togetherjava/tjbot/commands/help/AutoPruneHelperRoutine.java

```diff
@@ -0,0 +1,168 @@
+    private static final Logger logger = LoggerFactory.getLogger(AutoPruneHelperRoutine.class);
+
+    private static final int ROLE_FULL_LIMIT = 100;
+    private static final int ROLE_FULL_THRESHOLD = 95;
+    private static final int PRUNE_MEMBER_AMOUNT = 10;
+    private static final Period INACTIVE_AFTER = Period.ofDays(90);
+    private static final int RECENTLY_JOINED_DAYS = 7;
+
+    private final HelpSystemHelper helper;
+    private final ModAuditLogWriter modAuditLogWriter;
+    private final Database database;
+    private final List<String> allCategories;
+
+    /**
+     * Creates a new instance.
+     *
+     * @param config to determine all helper categories
+     * @param helper the helper to use
+     * @param modAuditLogWriter to inform mods when manual pruning becomes necessary
+     * @param database to determine whether an user is inactive
+     */
+    public AutoPruneHelperRoutine(@NotNull Config config, @NotNull HelpSystemHelper helper,
+            @NotNull ModAuditLogWriter modAuditLogWriter, @NotNull Database database) {
+        allCategories = config.getHelpSystem().getCategories();
+        this.helper = helper;
+        this.modAuditLogWriter = modAuditLogWriter;
+        this.database = database;
+    }
+
+    @Override
+    public @NotNull Schedule createSchedule() {
+        return new Schedule(ScheduleMode.FIXED_RATE, 0, 1, TimeUnit.HOURS);
+    }
+
+    @Override
+    public void runRoutine(@NotNull JDA jda) {
+        jda.getGuildCache().forEach(this::pruneForGuild);
+    }
+
+    private void pruneForGuild(@NotNull Guild guild) {
+        TextChannel overviewChannel = guild.getTextChannels()
+            .stream()
+            .filter(channel -> helper.isOverviewChannelName(channel.getName()))
+            .findAny()
+            .orElseThrow();
+        Instant when = Instant.now();
+
+        allCategories.stream()
+            .map(category -> helper.handleFindRoleForCategory(category, guild))
+            .filter(Optional::isPresent)
+            .map(Optional::orElseThrow)
+            .forEach(role -> pruneRoleIfFull(role, overviewChannel, when));
+    }
+
+    private void pruneRoleIfFull(@NotNull Role role, @NotNull TextChannel overviewChannel,
+            @NotNull Instant when) {
+        role.getGuild().findMembersWithRoles(role).onSuccess(members -> {
+            if (isRoleFull(members)) {
+                logger.debug("Helper role {} is full, starting to prune.", role.getName());
+                pruneRole(role, members, overviewChannel, when);
+            }
+        });
+    }
+
+    private boolean isRoleFull(@NotNull Collection<Member> members) {
+        return members.size() >= ROLE_FULL_THRESHOLD;
+    }
+
+    private void pruneRole(@NotNull Role role, @NotNull List<? extends Member> members,
+            @NotNull TextChannel overviewChannel, @NotNull Instant when) {
+        List<Member> membersShuffled = new ArrayList<>(members);
+        Collections.shuffle(membersShuffled);
```

Comment:
Why the shuffle?

Reply:
Because we decide to kick out, lets say 10 users out of 30 users eligible for the prune. I dont want this to happen alphabetically or anything, to not put anyone at a disadvantage.

So i am picking the people to prune at random